pkg/block/azure/adapter.go

```diff
@@ -196,6 +214,46 @@ func (a *Adapter) Get(ctx context.Context, obj block.ObjectPointer, _ int64) (io
 	return a.Download(ctx, obj, 0, azblob.CountToEnd)
 }
 
+func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer, mode block.PreSignMode) (string, error) {
+	permissions := azblob.BlobSASPermissions{Read: true}
+	if mode == block.PreSignModeWrite {
+		permissions = azblob.BlobSASPermissions{Write: true}
+	}
+	return a.getPreSignedURL(ctx, obj, permissions)
+}
+
+func (a *Adapter) getPreSignedURL(ctx context.Context, obj block.ObjectPointer, permissions azblob.BlobSASPermissions) (string, error) {
+	qualifiedKey, err := resolveBlobURLInfo(obj)
+	if err != nil {
+		return "", err
+	}
+	vals := azblob.BlobSASSignatureValues{
+		Protocol:      azblob.SASProtocolHTTPS, // Users MUST use HTTPS (not HTTP)
+		ExpiryTime:    a.preSignedURLDurationGenerator(),
+		ContainerName: qualifiedKey.ContainerName,
+		BlobName:      qualifiedKey.BlobURL,
+		Permissions:   permissions.String(),
+	}
+
+	keyCredentials, ok := a.credentials.(*azblob.SharedKeyCredential)
```

Comment:
Can be done once at `NewAdater` and leave here just pointer check.

Reply:
Done